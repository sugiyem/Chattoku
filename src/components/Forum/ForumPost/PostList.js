import { useEffect, useState } from "react";
import { SectionList } from "react-native";
import FetchPost from "./FetchPost";
import PostCard from "./PostCard";

const PostList = ({ forumId }) => {
  const [posts, setPosts] = useState([]);

  useEffect(() => {
    FetchPost(
      forumId,
      (data) => setPosts(data),
      () => {}
    );
  }, []);

  //
  console.log("POST LIST");
  console.log(posts);

  const renderItem = ({ section, item }) => {
    return <PostCard {...item} forumId={forumId} setPosts={setPosts} />;
  };

  const renderHeader = () => <></>;

  const renderFooter = () => <></>;

  return (
    <>
      <SectionList
        removeClippedSubviews={true}
        sections={[{ data: posts }]}
        renderItem={renderItem}
        renderHeader={renderHeader}
        renderFooter={renderFooter}
      />
    </>
  );
};

export default PostList;
