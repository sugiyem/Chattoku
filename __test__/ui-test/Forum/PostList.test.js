import React from "react";
import renderer from "react-test-renderer";
import PostList from "../../../src/components/Forum/ForumPost/PostList";

jest.mock("../../../src/services/Forum/HandleForumPost");
jest.mock("../../../src/services/Profile/FetchUserInfo");

describe("Test Post List UI", () => {
  it("Renders correctly", () => {
    const tree = renderer
      .create(
        <PostList
          forumId="forum-1"
          isBanned={false}
          isOwner={true}
          posts={[]}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
