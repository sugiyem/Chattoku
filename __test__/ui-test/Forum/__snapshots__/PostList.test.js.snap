// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Post List UI Renders correctly 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "data": Array [],
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={true}
  renderFooter={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={
    Array [
      0,
    ]
  }
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Search By Title..."
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "black",
            "borderRadius": 10,
            "borderWidth": 1,
            "color": "black",
            "flexGrow": 0,
            "flexShrink": 0,
            "margin": 10,
            "padding": 5,
          }
        }
      />
    </View>
    <View
      onLayout={[Function]}
      style={null}
    />
  </View>
</RCTScrollView>
`;
